kingpin/actors/rightscale/server_array.py

```diff
@@ -216,6 +257,48 @@ def _check_array_inputs(self, array, inputs):
         raise gen.Return()
 
     @gen.coroutine
+    def _update_params(self, array):
+        """Update the parameters on a RightScale ServerArray.
+
+        args:
+            array: The array to operate on
+        """
+
+        if not self.option('params'):
+            raise gen.Return()
+
+        params = self._generate_rightscale_params(
```

Comment:
`_generate_rigthscale_params` doesn't do any blocking operations, but it can raise exceptions. I think we should precompute the params and inputs inside of `__init__` for faster error catching.